### ISSUE TITLE
OTS: For BIP38 - there is a really fast 1040 keys per second repo on here too - https://github.com/agurodriguez/brute_force_bip38.git I think it's considerably faster than btcrecover, but it's also written in cpp, and is full of bugs.

### ISSUE THREAD

**Issue by sigkill** [3rdIteration/btcrecover#228]
Worth a look. The BIP38 routine in this repo is incredibly fast. I tested with a few test BIP38 encrypted wallet keys I generated, and it was able to find the correct password.

1040 keys per second on a dual Xeon 16 core each, about 3K on a 64 core cpu.

https://github.com/agurodriguez/brute_force_bip38.git I think it's considerably faster than btcrecover, but it's also written in cpp, and is full of bugs.

_Originally posted by @sigkill in https://github.com/3rdIteration/btcrecover/issues/227#issuecomment-896412279_

Just wanted to open a ticket incase it could be re-written and incorporated into btcrecover.py

**Comment by sigkill** [comment]
#32

**Comment by sigkill** [comment]
![image](https://user-images.githubusercontent.com/634740/129132446-4b08ff2b-144a-4741-8466-d45fcc0be9c2.png)

Cool this actually works. 
`--no-eta --enable-opencl  --opencl-platform 0   --opencl-workgroup-size 8 --threads=480`

**Comment by sigkill** [comment]
Didn't get any output stats, it locked up the mic0 and mic1 cards and rebooted them. There's also no flag to disable selected devices from what I can tell.

**Comment by 3rdIteration** [comment]
Open CL hanging on sCrypt means you are running out of memory.

In terms of performance, are you doing an apples to apples comparison in terms of wallet difficulty?

**Comment by sigkill** [comment]
Yeah same wallet on btcrecover and with the cpp based brute_force_bip38, the bforce loads and parses your entire password list (in this case the password is unknown, but I did validate with a known passphrase / test key) The configuration is a bit different too and you have to figure out the lowercase hex address to use the bforce script. It *almost* compiles on the Intel k1om based Xeon Phi, but errors when linking to libwally, which is a real shame because I think it would scream on that hardware.

But yeah - tested with a 64 core cpu and tested btcrecover against my known wallet, along with the brute_force_bip38, which was much faster.

There is a python libwally implementation that might do the trick, if the bip38 parts of btcrecover can by cython / cpp / c compiled. The performance comparison between the two is pretty substantial.
